src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -428,24 +428,48 @@ private string StripFormatSpecifier(string exp, out string formatSpecifier)
             }
 
             // array with static size
-            // TODO: could return '(T(*)[n])(exp)' but requires T
             var m = Regex.Match(trimmed, @"^\[?(\d+)\]?$");
             if (m.Success)
             {
+                string count = m.Groups[1].Value; // (\d+) capture group
+                string expr = exp.Substring(0, lastComma);
+
                 if (_engine.DebuggedProcess.MICommandFactory.Mode == MIMode.Gdb)
-                    return "*" + exp.Substring(0, lastComma) + "@" + trimmed;  // this does not work for lldb
+                {
+                    // return *<expression>@<count> which is only supported in GDB
+                    return $"*{expr}@{count}";
+                }
+                else
+                {
+                    string derefType = GetDereferencedTypeString(expr, threadId, frameLevel);
 
-                return exp.Substring(0, lastComma);
+                    // return '*(T(*)[n])(exp)'
+                    return $"*({derefType}(*)[{count}])({expr})";
```

Comment:
>(*) [](http://example.com/codeflow?start=42&length=3)

Do you understand why this is `(*)`?

Reply:
Not really sure, but my guess is casting it to an array of T breaks with `C-style cast from 'int' to 'int [10]' is not allowed`, but if we cast the `exp` to a function that returns an `array with size 'n' of T elements` it works.

Reply:
This is what I was using for reference to clarify the `(*)`.
https://stackoverflow.com/a/48133012

Reply:
For clarity, this isn't a pointer to a function, but rather a pointer to an array. Weird syntax, but I guess it makes sense.